### ISSUE TITLE
Limited maximum number of particles

### ISSUE THREAD

**Issue by jccurtis** [zoglauer/megalib#23]
I can not run a `PtSrc` with greater than about 2.1e9 particles. I receive the following error from `cosima -v 3 PtResp.source`:

```
Adding run PointSource
Setting output file name PtResp for run PointSource

   ***  Error  ***  in parsed file /Users/joey/odrive/LBLDrive/misc/2017-06_megalib_nparticles_bug/PtResp.source at line 16:
   ***  Error  ***  Can not parse token Events correctly: Number not positive?
   Line (3 tokens): "PointSource.Events       5e9"

MParser: Unable to parse file "PtResp.source"
Error: Unable to parse parameter file "PtResp.source" correctly!
An error occurred during initialization
```

As far as I can tell, the `MTokenizer` [here](https://github.com/zoglauer/megalib/blob/master/src/global/misc/src/MTokenizer.cxx#L227-L245) is returning `0` and then the `MCParameterFile` raises an error [here](https://github.com/zoglauer/megalib/blob/master/src/cosima/src/MCParameterFile.cc#L691-L700). I don't know how to access the `mlog` information. 

I feel like I'm missing something and my knowledge of c++ is limited. I recognize that 2.1e9 ~= 2**31 but the number of particles is returned as a `double`, so I would think this shouldn't matter. That said, 3e9 particles doesn't work so I think the issue is related to an `unsigned int` somewhere. I'd be happy to make a patch but I'm not sure what is going wrong.

I've attached a zip of the relevant files. I've tested this on macOS and Ubuntu (16.04). I'm running a slightly older version of MEGAlib (`2.29.05`). Thanks!

[2017-06_megalib_nparticles_bug.zip](https://github.com/zoglauer/megalib/files/1053527/2017-06_megalib_nparticles_bug.zip)

**Comment by zoglauer** [comment]
Thanks a lot for reporting this.

The ".Events" has to be an integer and not a double. I added a protection in the latest version (branch: experimental).

Actually Geant4 cannot simulate more than 2^31 events in one run, because its event ID is an int and not a long. In the development version (branch: "experimental"), I have added some code to circumvent that, since we encountered the same issue with COSI, but it is not yet working perfectly...

For the time being, just make a few smaller runs and add them up.
